modules/account/src/Volo.Abp.Account.Web/Pages/Account/Login.cshtml.cs

```diff
@@ -211,6 +211,7 @@ protected virtual async Task<IdentityUser> CreateExternalUserAsync(ExternalLogin
 
             CheckIdentityErrors(await UserManager.CreateAsync(user));
             CheckIdentityErrors(await UserManager.SetEmailAsync(user, emailAddress));
+            CheckIdentityErrors(await UserManager.SetRolesAsync(user, RoleManager.Roles.Where(r => r.IsDefault).Select(r => r.Name)));
```

Comment:
Don't use RoleManager.Roles.Where... which executes a query to the database synchronously (because now way to make it async with this code)

Instead, do it just like the AccountAppService:
https://github.com/abpframework/abp/blob/ecac0358fbc9a0acbbda91111d41d5ee6bfb795a/modules/account/src/Volo.Abp.Account.Application/Volo/Abp/Account/AccountAppService.cs#L39-L44

Reply:
How to get the default role in the web layer?

Should we add two endpoints to the Role application service to get all the `static/default` roles. like https://github.com/abpframework/abp/pull/2985

Or in `RoleManager`?

Reply:
I've done it with https://github.com/abpframework/abp/commit/99f23a51bc039d4de5d36d821476be73e8536175